devel/tilt/Portfile

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           golang 1.0
+
+go.setup            github.com/tilt-dev/tilt 0.18.1 v
+revision            0
+
+categories          devel
+maintainers         nomaintainer
+platforms           darwin
+supported_archs     x86_64
+license             Apache-2
+
+description         Run Kubernetes environments locally, for local development
+
+long_description    Tilt powers multi-service development and makes sure they \
+                    behave! Run tilt up to work in a complete dev environment \
+                    configured for your team. \
+                    \
+                    Tilt automates all the steps from a code change to a new \
+                    process: watching files, building container images, and \
+                    bringing your environment up-to-date. Think docker build \
+                    && kubectl apply or docker-compose up.
+
+homepage            https://tilt.dev/
+
+checksums           rmd160  d2bd6001e581f54986011e10f44a58b68b6f2f80 \
+                    sha256  1a4cb78c4e880157d22d6b8e209d7fc70e77dfef0136929047e5eeee96b6841c \
+                    size    20524516
+
+depends_build       port:yarn port:nodejs12
+use_parallel_build  no
+build.cmd           "make build-js && ${go.bin} build -mod vendor ./cmd/tilt"
+build.env-replace    GO111MODULE=off GO111MODULE=on
```

Comment:
I completely understand if this is required, but it runs into https://trac.macports.org/ticket/61192. It would be preferable to use `go.vendors`, but please correct me if this is not viable for this port.

(This doesn't matter as much anyway though since the JS deps have no choice but to be downloaded at build time)

Reply:
I had tried the port2go thingy, but there was at least one unsupported source repo (google's repo for protobuf), and I suspect there were several more. There was roughly 1000 lines of golang deps!

Reply:
I can leave a comment explaining the history

Reply:
Just looking through it, it should be possible.

Here are a few things I noticed. There will be more manual changes that will need to be made, but this should help:

|Original   |  New |  Notes |
|---|---|---|
|  go.googlesource.com/protobuf |  github.com/protocolbuffers/protobuf-go |  Set using repo |
|  gopkg.in/fsnotify.v1  |  github.com/fsnotify/fsnotify |  Set using repo |
|  v17.12.0-ce-rc1.0.20181016164535-5271c7cb484d+incompatible  |  v17.12.0-ce-rc1 |  Set using lock for docker/docker. You might have to use repo to set it to moby/moby |

I would recommend looking at the beginning comments of the [golang portgroup](https://github.com/macports/macports-ports/blob/d79aad1b301412f3ff83438f4e0c23d3c513aeb2/_resources/port1.0/group/golang-1.0.tcl) to help. Let me know if anything's not clear and I'll try and assist you :)

Reply:
I'm going to push up a WIP. After a bunch of hacking on the go.vendors I have everything downloading and verifying, but the pre-checkout GOPATH setup stuff in the golang portgroup is failing. It seems to be due to edge cases with identical versions of packages installed under different names.